internal/pkg/otel/run.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package otel
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/converter/expandconverter"
+	"go.opentelemetry.io/collector/otelcol"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/elastic-agent/internal/pkg/release"
+)
+
+const buildDescription = "Elastic opentelemetry-collector distribution"
+
+func IsOtelConfig(ctx context.Context, pathConfigFile string) (bool, error) {
+	fileName := filepath.Base(pathConfigFile)
+	cleanFileName := strings.TrimSpace(strings.ToLower(strings.TrimSuffix(fileName, filepath.Ext(fileName))))
+	if cleanFileName == "otel" || cleanFileName == "otlp" || cleanFileName == "otelcol" {
+		return true, nil
+	}
```

Comment:
IIUC, the only expected extensions are `.yml` or `.yaml`.  Could we be a bit stricter here and only accept those extensions?

Reply:
I don't think this has been implemented yet?  If this code sees `otel.json` it will return `(true, nil)`, right?